### PR TITLE
Refactor edd_object_to_array 

### PR DESCRIPTION
**Pull request by cklosowski**
#4042

### CODE REVIEW

File: includes/misc-functions.php [hunk 1]
```diff
@@ -904,3 +904,29 @@ function edd_can_view_receipt( $payment_key = '' ) {
...
 }
+
+/**
+ * Given an object or array of objects, convert them to arrays
+ *
+ * @since  2.6
+ * @param  object|array $object An object or an array of objects
+ * @return array                An array or array of arrays, converted from the provided object(s)
+ */
+function edd_array_convert( $object = array() ) {
```
Comment (pippinsplugins) [inline includes/misc-functions.php]: Can we please call this `edd_oject_to_array()`?

Comment (cklosowski) [inline includes/misc-functions.php]: Yep!

**Comment by cklosowski** [comment]
@pippinsplugins ok updated to just refactor our existing function

File: tests/tests-misc.php [hunk 2]
```diff
@@ -614,4 +614,24 @@ public function test_cart_url_formats() {
 
 		EDD_Helper_Download::delete_download( $post->ID );
 	}
+
+	public function test_array_convert() {
+		$customer1_id = EDD()->customers->add( array( 'email' => 'test10@example.com' ) );
```
Comment (pippinsplugins) [inline tests/tests-misc.php]: These tests should be expanded to check for various properties inside of the object. That will help prevent against this: https://github.com/easydigitaldownloads/easy-digital-downloads/issues/4042#issuecomment-220081416

Comment (cklosowski) [inline tests/tests-misc.php]: I'll add a few more checks on the customer object, email, and purchase count.

Comment (cklosowski) [inline tests/tests-misc.php]: @pippinsplugins good with those updates (below?)

Comment (pippinsplugins) [inline tests/tests-misc.php]: Yep, working great for me.

**Pull request merged**